astropy/wcs/src/wcslib_wrap.c

```diff
@@ -1621,14 +1622,36 @@ PyWcsprm_cset(
     const int convert) {
 
   int status = 0;
+  int cmp = 0;
+  double tolerance = 0;
+  int equal;
+
+  // We want to avoid calling wcsset whenever possible as it is not thread-safe,
+  // so we keep track of what the wcsprm was last time this function was called,
+  // using x_prev, and we compare these. If equal, and if PyWcsprm_cset() succeeded
+  // last time it was called, then we can just return 0 as there should be nothing
+  // to do.
+  if(self->x_prev_set) {
```

Comment:
self->x_prev_set needs to be initialised somewhere in the init (perhaps where `x.flag` is initialised) otherwise this could potentially be an use of uninitialised memory